The job **`x86_64-gnu-integration`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/8770333034/job/24066804563) [(plain)](https://github.com/rust-lang-ci/rust/commit/b6bd9391b77993f8828c1d7e01e004979b36bbeb/checks/24066804563/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

warning: 8 warnings emitted


FAILED: host_x64/gen/src/storage/fxfs/lib.clippy 
../../build/rust/clippy_wrapper.sh --output host_x64/gen/src/storage/fxfs/lib.clippy --jq ../../prebuilt/third_party/jq/linux-x64/bin/jq --fail -- env  ../../../install/bin/clippy-driver ../../src/storage/fxfs/src/lib.rs --sysroot=../../../install --crate-type=rlib --cfg=__rust_toolchain=\"2024/04/21\ 05:43:23\" -Clinker=../../prebuilt/third_party/clang/linux-x64/bin/clang++ -Cdefault-linker-libraries -Clink-arg=--sysroot=../../prebuilt/third_party/sysroot/linux -Clink-arg=--target=x86_64-unknown-linux-gnu -Clink-arg=-stdlib=libc++ -Clink-arg=--start-no-unused-arguments -Clink-arg=-unwindlib=libunwind -Clink-arg=-rtlib=compiler-rt -Clink-arg=--end-no-unused-arguments -Clink-arg=-fuse-ld=lld -Clink-arg=-Wl,--build-id -Zremap-cwd-prefix=. -Copt-level=0 -Cdebuginfo=0 -Clink-arg=-gz=zstd -Cdebug-assertions=yes -Zallow-features= --target x86_64-unknown-linux-gnu --cap-lints=warn -Wrust-2018-idioms -Dderef-nullptr -Dinvalid-value -Dunused-must-use -Awhere_clauses_object_safety -Dwarnings -Clinker=../../prebuilt/third_party/clang/linux-x64/bin/clang++ -Cdefault-linker-libraries -Clink-arg=--sysroot=../../prebuilt/third_party/sysroot/linux -Clink-arg=--target=x86_64-unknown-linux-gnu -Clink-arg=-stdlib=libc++ -Clink-arg=--start-no-unused-arguments -Clink-arg=-unwindlib=libunwind -Clink-arg=-rtlib=compiler-rt -Clink-arg=--end-no-unused-arguments -Clink-arg=-fuse-ld=lld -Clink-arg=-Wl,--build-id -Clink-arg=-fPIE -Clink-arg=-pie -Clink-arg=-static-libstdc++ -Clink-arg=-Wl,-rpath=\$ORIGIN/ -Wunused_crate_dependencies -Dunused_results -Aunused_results --edition=2021 -Aclippy::all -Dclippy::correctness -Dclippy::missing_safety_doc -Wclippy::expect_fun_call -Wclippy::unnecessary_mut_passed --cfg=fidl_rust_transition_step=\"migrate_all_results\"
error[E0271]: expected `impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send` to be a future that resolves to `Result<impl LayerIterator<K, V>, Error>`, but it resolves to `Result<impl LayerIterator<K, V>, Error>`
   |
   |
27 |     ) -> impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send {
   |
   |
   = note: expected enum `std::result::Result<impl lsm_tree::types::LayerIterator<K, V>, _>` (opaque type at <../../src/storage/fxfs/src/object_store/tree.rs:27:38>)
              found enum `std::result::Result<impl lsm_tree::types::LayerIterator<K, V>, _>` (trait associated opaque type at <../../src/storage/fxfs/src/object_store/tree.rs:27:38>)
note: required by a bound in `object_store::tree::MajorCompactable::{synthetic#0}`
   |
   |
27 |     ) -> impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send {
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `MajorCompactable::{synthetic#0}`

error: aborting due to 1 previous error



FAILED: host_x64/obj/src/storage/fxfs/libfxfs.rlib 
../../prebuilt/third_party/python3/linux-x64/bin/python3 -S ../../build/tracer/restat_cacher.py --outputs host_x64/obj/src/storage/fxfs/libfxfs.rlib host_x64/obj/src/storage/fxfs/libfxfs.rlib.d --  RUST_BACKTRACE=1 ../../../install/bin/rustc --color=always --crate-name fxfs ../../src/storage/fxfs/src/lib.rs --crate-type rlib --emit=dep-info=host_x64/obj/src/storage/fxfs/libfxfs.rlib.d,link -Zdep-info-omit-d-target --cfg=fidl_rust_transition_step=\"migrate_all_results\" --cfg=__rust_toolchain=\"2024/04/21\ 05:43:23\" -Clinker=../../prebuilt/third_party/clang/linux-x64/bin/clang++ -Cdefault-linker-libraries -Clink-arg=--sysroot=../../prebuilt/third_party/sysroot/linux -Clink-arg=--target=x86_64-unknown-linux-gnu -Clink-arg=-stdlib=libc++ -Clink-arg=--start-no-unused-arguments -Clink-arg=-unwindlib=libunwind -Clink-arg=-rtlib=compiler-rt -Clink-arg=--end-no-unused-arguments -Clink-arg=-fuse-ld=lld -Clink-arg=-Wl,--build-id -Zremap-cwd-prefix=. -Copt-level=0 -Cdebuginfo=0 -Clink-arg=-gz=zstd -Cdebug-assertions=yes -Zallow-features= --target x86_64-unknown-linux-gnu --cap-lints=warn -Wrust-2018-idioms -Dderef-nullptr -Dinvalid-value -Dunused-must-use -Awhere_clauses_object_safety -Dwarnings -Clinker=../../prebuilt/third_party/clang/linux-x64/bin/clang++ -Cdefault-linker-libraries -Clink-arg=--sysroot=../../prebuilt/third_party/sysroot/linux -Clink-arg=--target=x86_64-unknown-linux-gnu -Clink-arg=-stdlib=libc++ -Clink-arg=--start-no-unused-arguments -Clink-arg=-unwindlib=libunwind -Clink-arg=-rtlib=compiler-rt -Clink-arg=--end-no-unused-arguments -Clink-arg=-fuse-ld=lld -Clink-arg=-Wl,--build-id -Clink-arg=-fPIE -Clink-arg=-pie -Clink-arg=-static-libstdc++ -Clink-arg=-Wl,-rpath=\$ORIGIN/ -Wunused_crate_dependencies -Dunused_results -Aunused_results --edition=2021 -o host_x64/obj/src/storage/fxfs/libfxfs.rlib -Ldependency=host_x64/obj/src/storage/fxfs -Ldependency=host_x64/obj/sdk/fidl/fuchsia.io/fuchsia.io_rust -Ldependency=host_x64/obj/src/lib/fidl/rust/fidl -Ldependency=host_x64/obj/src/lib/fuchsia-async -Ldependency=host_x64/obj/src/lib/fuchsia-async-macro -Ldependency=host_x64/obj/third_party/rust_crates -Ldependency=host_x64/obj/src/lib/zircon/rust -Ldependency=host_x64/obj/src/lib/fuchsia-sync -Ldependency=host_x64/obj/sdk/fidl/fuchsia.unknown/fuchsia.unknown_rust -Ldependency=host_x64/obj/src/lib/diagnostics/inspect/rust -Ldependency=host_x64/obj/src/lib/diagnostics/hierarchy/rust -Ldependency=host_x64/obj/sdk/fidl/fuchsia.diagnostics/fuchsia.diagnostics_rust -Ldependency=host_x64/obj/sdk/fidl/fuchsia.mem/fuchsia.mem_rust -Ldependency=host_x64/obj/src/lib/diagnostics/selectors -Ldependency=host_x64/obj/src/sys/lib/moniker -Ldependency=host_x64/obj/src/sys/lib/cm_rust -Ldependency=host_x64/obj/sdk/fidl/fuchsia.component.decl/fuchsia.component.decl_rust -Ldependency=host_x64/obj/sdk/fidl/fuchsia.data/fuchsia.data_rust -Ldependency=host_x64/obj/sdk/fidl/fuchsia.url/fuchsia.url_rust -Ldependency=host_x64/obj/sdk/fidl/fuchsia.process/fuchsia.process_rust -Ldependency=host_x64/obj/sdk/fidl/fuchsia.ldsvc/fuchsia.ldsvc_rust -Ldependency=host_x64/obj/src/lib/flyweights -Ldependency=host_x64/obj/src/lib/from-enum -Ldependency=host_x64/obj/src/sys/lib/cm_fidl_validator -Ldependency=host_x64/obj/src/lib/directed_graph -Ldependency=host_x64/obj/src/sys/lib/cm_types -Ldependency=host_x64/obj/src/storage/lib/vfs/rust/name -Ldependency=host_x64/obj/src/sys/lib/namespace -Ldependency=host_x64/obj/sdk/fidl/fuchsia.component/fuchsia.component_rust -Ldependency=host_x64/obj/sdk/fidl/fuchsia.component.sandbox/fuchsia.component.sandbox_rust -Ldependency=host_x64/obj/sdk/fidl/fuchsia.component.runner/fuchsia.component.runner_rust -Ldependency=host_x64/obj/sdk/fidl/fuchsia.diagnostics.types/fuchsia.diagnostics.types_rust -Ldependency=host_x64/obj/src/sys/lib/bitflags-serde-legacy -Ldependency=host_x64/obj/src/lib/diagnostics/inspect/format/rust -Ldependency=host_x64/obj/src/lib/injectable-time -Ldependency=host_x64/obj/src/lib/mundane -Ldependency=host_x64/obj/third_party/boringssl/rust/boringssl-sys -Ldependency=host_x64/obj/src/storage/fxfs/crypto -Ldependency=host_x64/obj/src/storage/fxfs/trace -Ldependency=host_x64/obj/src/storage/fxfs/trace/macros -Ldependency=host_x64/obj/src/storage/lib/trace -Ldependency=host_x64/obj/src/storage/type-fprint -Ldependency=host_x64/obj/src/storage/lib/fsverity-merkle -Ldependency=host_x64/obj/src/storage/lib/storage_device -Lnative=host_x64/obj/third_party/boringssl -Clink-arg=-Bdynamic -Clink-arg=host_x64/obj/third_party/boringssl/libcrypto.a --extern fxfs_macros=host_x64/obj/src/storage/fxfs/libfxfs_macros.so --extern fidl_fuchsia_io=host_x64/obj/sdk/fidl/fuchsia.io/fuchsia.io_rust/libfidl_fuchsia_io.rlib --extern fuchsia_async=host_x64/obj/src/lib/fuchsia-async/libfuchsia_async.rlib --extern futures=host_x64/obj/third_party/rust_crates/libfutures-e6516813fcd2ea59.rlib --extern static_assertions=host_x64/obj/third_party/rust_crates/libstatic_assertions-5fff9c36214c34b7.rlib --extern byteorder=host_x64/obj/third_party/rust_crates/libbyteorder-eb0270683cf7e0fd.rlib --extern serde=host_x64/obj/third_party/rust_crates/libserde-703d0860e891586d.rlib --extern event_listener=host_x64/obj/third_party/rust_crates/libevent_listener-db0b634d4cbcc40b.rlib --extern thiserror=host_x64/obj/third_party/rust_crates/libthiserror-dcdc66d3bbb690d3.rlib --extern scopeguard=host_x64/obj/third_party/rust_crates/libscopeguard-6ef302a7f864fbb3.rlib --extern tracing=host_x64/obj/third_party/rust_crates/libtracing-979babe725aa0770.rlib --extern once_cell=host_x64/obj/third_party/rust_crates/libonce_cell-283c53dba5ff9f5a.rlib --extern fuchsia_inspect=host_x64/obj/src/lib/diagnostics/inspect/rust/libfuchsia_inspect.rlib --extern either=host_x64/obj/third_party/rust_crates/libeither-f4134ec734cf166b.rlib --extern anyhow=host_x64/obj/third_party/rust_crates/libanyhow-99a34437cafed13a.rlib --extern num_traits=host_x64/obj/third_party/rust_crates/libnum_traits-eace97eec9f18ef0.rlib --extern async_trait=host_x64/obj/third_party/rust_crates/libasync_trait-1fd7cf147dfbb38f.so --extern mundane=host_x64/obj/src/lib/mundane/libmundane.rlib --extern fxfs_crypto=host_x64/obj/src/storage/fxfs/crypto/libfxfs_crypto.rlib --extern fxfs_trace=host_x64/obj/src/storage/fxfs/trace/libfxfs_trace.rlib --extern fprint=host_x64/obj/src/storage/type-fprint/libfprint.rlib --extern fsverity_merkle=host_x64/obj/src/storage/lib/fsverity-merkle/libfsverity_merkle.rlib --extern storage_device=host_x64/obj/src/storage/lib/storage_device/libstorage_device.rlib --extern rand=host_x64/obj/third_party/rust_crates/librand-3acdc77c21a3a508.rlib --extern assert_matches=host_x64/obj/third_party/rust_crates/libassert_matches-568351bf14d9d02.rlib --extern bincode=host_x64/obj/third_party/rust_crates/libbincode-2294ada8ceec0c61.rlib --extern linked_hash_map=host_x64/obj/third_party/rust_crates/liblinked_hash_map-b9b35caede8d8189.rlib --extern rustc_hash=host_x64/obj/third_party/rust_crates/librustc_hash-d18fd5c05c7b0ca4.rlib --extern uuid=host_x64/obj/third_party/rust_crates/libuuid-10961005f47a1748.rlib && ../../prebuilt/third_party/python3/linux-x64/bin/python3 -S ../../build/gn/verify_depfile.py -t "//src/storage/fxfs:lib.actual(//build/toolchain:host_x64)" -d host_x64/obj/src/storage/fxfs/libfxfs.rlib.d ../../src/storage/fxfs/src/checksum.rs ../../src/storage/fxfs/src/debug_assert_not_too_long.rs ../../src/storage/fxfs/src/drop_event.rs ../../src/storage/fxfs/src/errors.rs ../../src/storage/fxfs/src/filesystem.rs ../../src/storage/fxfs/src/fsck.rs ../../src/storage/fxfs/src/fsck/errors.rs ../../src/storage/fxfs/src/fsck/store_scanner.rs ../../src/storage/fxfs/src/fsck/tests.rs ../../src/storage/fxfs/src/lib.rs ../../src/storage/fxfs/src/log.rs ../../src/storage/fxfs/src/lsm_tree.rs ../../src/storage/fxfs/src/lsm_tree/cache.rs ../../src/storage/fxfs/src/lsm_tree/merge.rs ../../src/storage/fxfs/src/lsm_tree/simple_persistent_layer.rs ../../src/storage/fxfs/src/lsm_tree/skip_list_layer.rs ../../src/storage/fxfs/src/lsm_tree/types.rs ../../src/storage/fxfs/src/metrics.rs ../../src/storage/fxfs/src/object_handle.rs ../../src/storage/fxfs/src/object_store.rs ../../src/storage/fxfs/src/object_store/allocator.rs ../../src/storage/fxfs/src/object_store/allocator/merge.rs ../../src/storage/fxfs/src/object_store/caching_object_handle.rs ../../src/storage/fxfs/src/object_store/data_object_handle.rs ../../src/storage/fxfs/src/object_store/directory.rs ../../src/storage/fxfs/src/object_store/extent_record.rs ../../src/storage/fxfs/src/object_store/flush.rs ../../src/storage/fxfs/src/object_store/graveyard.rs ../../src/storage/fxfs/src/object_store/journal.rs ../../src/storage/fxfs/src/object_store/journal/bootstrap_handle.rs ../../src/storage/fxfs/src/object_store/journal/checksum_list.rs ../../src/storage/fxfs/src/object_store/journal/reader.rs ../../src/storage/fxfs/src/object_store/journal/super_block.rs ../../src/storage/fxfs/src/object_store/journal/writer.rs ../../src/storage/fxfs/src/object_store/key_manager.rs ../../src/storage/fxfs/src/object_store/merge.rs ../../src/storage/fxfs/src/object_store/object_manager.rs ../../src/storage/fxfs/src/object_store/object_record.rs ../../src/storage/fxfs/src/object_store/project_id.rs ../../src/storage/fxfs/src/object_store/store_object_handle.rs ../../src/storage/fxfs/src/object_store/transaction.rs ../../src/storage/fxfs/src/object_store/tree.rs ../../src/storage/fxfs/src/object_store/tree_cache.rs ../../src/storage/fxfs/src/object_store/volume.rs ../../src/storage/fxfs/src/range.rs ../../src/storage/fxfs/src/round.rs ../../src/storage/fxfs/src/serialized_types.rs ../../src/storage/fxfs/src/serialized_types/test_traits.rs ../../src/storage/fxfs/src/serialized_types/tests.rs ../../src/storage/fxfs/src/serialized_types/traits.rs ../../src/storage/fxfs/src/serialized_types/types.rs ../../src/storage/fxfs/src/testing.rs ../../src/storage/fxfs/src/testing/fake_object.rs ../../src/storage/fxfs/src/testing/writer.rs ../../prebuilt/third_party/python3/linux-x64/bin/python3 ../../build/tracer/restat_cacher.py
error[E0271]: expected `impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send` to be a future that resolves to `Result<impl LayerIterator<K, V>, Error>`, but it resolves to `Result<impl LayerIterator<K, V>, Error>`
   |
   |
27 |     ) -> impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send {
   |
   |
   = note: expected enum `Result<impl LayerIterator<K, V>, _>` (opaque type at <../../src/storage/fxfs/src/object_store/tree.rs:27:38>)
              found enum `Result<impl LayerIterator<K, V>, _>` (trait associated opaque type at <../../src/storage/fxfs/src/object_store/tree.rs:27:38>)
note: required by a bound in `MajorCompactable::{synthetic#0}`
   |
   |
27 |     ) -> impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send {
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `MajorCompactable::{synthetic#0}`
error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0271`.
warning: constant `REBOOT_MANUALLY` is never used
---
    |    ^^^^^^^^^^^^^^^^^^^

warning: 10 warnings emitted

FAILED: host_x64/gen/src/storage/fxfs/lib_test_executable.clippy 
../../build/rust/clippy_wrapper.sh --output host_x64/gen/src/storage/fxfs/lib_test_executable.clippy --jq ../../prebuilt/third_party/jq/linux-x64/bin/jq --fail -- env  ../../../install/bin/clippy-driver ../../src/storage/fxfs/src/lib.rs --sysroot=../../../install --crate-type=bin --cfg=__rust_toolchain=\"2024/04/21\ 05:43:23\" -Clinker=../../prebuilt/third_party/clang/linux-x64/bin/clang++ -Cdefault-linker-libraries -Clink-arg=--sysroot=../../prebuilt/third_party/sysroot/linux -Clink-arg=--target=x86_64-unknown-linux-gnu -Clink-arg=-stdlib=libc++ -Clink-arg=--start-no-unused-arguments -Clink-arg=-unwindlib=libunwind -Clink-arg=-rtlib=compiler-rt -Clink-arg=--end-no-unused-arguments -Clink-arg=-fuse-ld=lld -Clink-arg=-Wl,--build-id -Zremap-cwd-prefix=. -Copt-level=0 -Cdebuginfo=0 -Clink-arg=-gz=zstd -Cdebug-assertions=yes -Zallow-features= --target x86_64-unknown-linux-gnu --cap-lints=warn -Wrust-2018-idioms -Dderef-nullptr -Dinvalid-value -Dunused-must-use -Awhere_clauses_object_safety -Dwarnings -Clinker=../../prebuilt/third_party/clang/linux-x64/bin/clang++ -Cdefault-linker-libraries -Clink-arg=--sysroot=../../prebuilt/third_party/sysroot/linux -Clink-arg=--target=x86_64-unknown-linux-gnu -Clink-arg=-stdlib=libc++ -Clink-arg=--start-no-unused-arguments -Clink-arg=-unwindlib=libunwind -Clink-arg=-rtlib=compiler-rt -Clink-arg=--end-no-unused-arguments -Clink-arg=-fuse-ld=lld -Clink-arg=-Wl,--build-id -Clink-arg=-fPIE -Clink-arg=-pie -Clink-arg=-static-libstdc++ -Clink-arg=-Wl,-rpath=\$ORIGIN/ -Wunused_crate_dependencies -Dunused_results -Aunused_results --edition=2021 -Aclippy::all -Dclippy::correctness -Dclippy::missing_safety_doc -Wclippy::expect_fun_call -Wclippy::unnecessary_mut_passed --cfg=fidl_rust_transition_step=\"migrate_all_results\" --test
error[E0271]: expected `impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send` to be a future that resolves to `Result<impl LayerIterator<K, V>, Error>`, but it resolves to `Result<impl LayerIterator<K, V>, Error>`
   |
   |
27 |     ) -> impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send {
   |
   |
   = note: expected enum `std::result::Result<impl lsm_tree::types::LayerIterator<K, V>, _>` (opaque type at <../../src/storage/fxfs/src/object_store/tree.rs:27:38>)
              found enum `std::result::Result<impl lsm_tree::types::LayerIterator<K, V>, _>` (trait associated opaque type at <../../src/storage/fxfs/src/object_store/tree.rs:27:38>)
note: required by a bound in `object_store::tree::MajorCompactable::{synthetic#0}`
   |
   |
27 |     ) -> impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send {
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `MajorCompactable::{synthetic#0}`

error: aborting due to 1 previous error


---
    = note: this warning originates in the derive macro `ToPrimitive` (in Nightly builds, run with -Z macro-backtrace for more info)

warning: 2 warnings emitted

FAILED: gen/src/storage/fxfs/_fuzz_journal_bytes_lib_rustc_static.clippy 
../../build/rust/clippy_wrapper.sh --output gen/src/storage/fxfs/_fuzz_journal_bytes_lib_rustc_static.clippy --jq ../../prebuilt/third_party/jq/linux-x64/bin/jq --fail -- env  ../../../install/bin/clippy-driver ../../src/storage/fxfs/src/lib.rs --sysroot=../../../install --crate-type=staticlib --cfg=fuzz --cfg=fuzz_target=\"fuzz_journal_bytes\" -Ccodegen-units=1 --cfg=__rust_toolchain=\"2024/04/21\ 05:43:23\" -L gen/zircon/public/sysroot/cpp/lib -Clinker=../../prebuilt/third_party/clang/linux-x64/bin/lld -Clink-arg=--sysroot=gen/zircon/public/sysroot/cpp -Clink-arg=-L../../prebuilt/third_party/clang/linux-x64/bin/../lib/x86_64-unknown-fuchsia/noexcept -Clink-arg=-L../../prebuilt/third_party/clang/linux-x64/lib/clang/18/lib/x86_64-unknown-fuchsia -Clink-arg=--pack-dyn-relocs=relr -Clink-arg=-dynamic-linker=ld.so.1 -Clink-arg=--icf=all -Clink-arg=-zrel -Zremap-cwd-prefix=. -Cforce-frame-pointers -Copt-level=0 -Cdebuginfo=0 -Clink-arg=--compress-debug-sections=zstd -Cdebug-assertions=yes -Zallow-features= --target x86_64-fuchsia --cap-lints=warn -Wrust-2018-idioms -Dderef-nullptr -Dinvalid-value -Dunused-must-use -Awhere_clauses_object_safety -Dwarnings -Cprefer-dynamic -Clink-args=-zstack-size=0x200000 -Cpanic=abort -Cforce-unwind-tables=yes -Zpanic_abort_tests -Csymbol-mangling-version=v0 -Wunused_crate_dependencies -Dunused_results -Aunused_results --edition=2021 -Aclippy::all -Dclippy::correctness -Dclippy::missing_safety_doc -Wclippy::expect_fun_call -Wclippy::unnecessary_mut_passed --cfg=fidl_rust_transition_step=\"migrate_all_results\"
error[E0271]: expected `impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send` to be a future that resolves to `Result<impl LayerIterator<K, V>, Error>`, but it resolves to `Result<impl LayerIterator<K, V>, Error>`
   |
   |
27 |     ) -> impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send {
   |
   |
   = note: expected enum `std::result::Result<impl lsm_tree::types::LayerIterator<K, V>, _>` (opaque type at <../../src/storage/fxfs/src/object_store/tree.rs:27:38>)
              found enum `std::result::Result<impl lsm_tree::types::LayerIterator<K, V>, _>` (trait associated opaque type at <../../src/storage/fxfs/src/object_store/tree.rs:27:38>)
note: required by a bound in `object_store::tree::MajorCompactable::{synthetic#0}`
   |
   |
27 |     ) -> impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send {
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `MajorCompactable::{synthetic#0}`

error: aborting due to 1 previous error



FAILED: gen/src/storage/fxfs/_fuzz_lsm_tree_actions_lib_rustc_static.clippy 
../../build/rust/clippy_wrapper.sh --output gen/src/storage/fxfs/_fuzz_lsm_tree_actions_lib_rustc_static.clippy --jq ../../prebuilt/third_party/jq/linux-x64/bin/jq --fail -- env  ../../../install/bin/clippy-driver ../../src/storage/fxfs/src/lib.rs --sysroot=../../../install --crate-type=staticlib --cfg=fuzz --cfg=fuzz_target=\"fuzz_lsm_tree_actions\" -Ccodegen-units=1 --cfg=__rust_toolchain=\"2024/04/21\ 05:43:23\" -L gen/zircon/public/sysroot/cpp/lib -Clinker=../../prebuilt/third_party/clang/linux-x64/bin/lld -Clink-arg=--sysroot=gen/zircon/public/sysroot/cpp -Clink-arg=-L../../prebuilt/third_party/clang/linux-x64/bin/../lib/x86_64-unknown-fuchsia/noexcept -Clink-arg=-L../../prebuilt/third_party/clang/linux-x64/lib/clang/18/lib/x86_64-unknown-fuchsia -Clink-arg=--pack-dyn-relocs=relr -Clink-arg=-dynamic-linker=ld.so.1 -Clink-arg=--icf=all -Clink-arg=-zrel -Zremap-cwd-prefix=. -Cforce-frame-pointers -Copt-level=0 -Cdebuginfo=0 -Clink-arg=--compress-debug-sections=zstd -Cdebug-assertions=yes -Zallow-features= --target x86_64-fuchsia --cap-lints=warn -Wrust-2018-idioms -Dderef-nullptr -Dinvalid-value -Dunused-must-use -Awhere_clauses_object_safety -Dwarnings -Cprefer-dynamic -Clink-args=-zstack-size=0x200000 -Cpanic=abort -Cforce-unwind-tables=yes -Zpanic_abort_tests -Csymbol-mangling-version=v0 -Wunused_crate_dependencies -Dunused_results -Aunused_results --edition=2021 -Aclippy::all -Dclippy::correctness -Dclippy::missing_safety_doc -Wclippy::expect_fun_call -Wclippy::unnecessary_mut_passed --cfg=fidl_rust_transition_step=\"migrate_all_results\"
error[E0271]: expected `impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send` to be a future that resolves to `Result<impl LayerIterator<K, V>, Error>`, but it resolves to `Result<impl LayerIterator<K, V>, Error>`
   |
   |
27 |     ) -> impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send {
   |
   |
   = note: expected enum `std::result::Result<impl lsm_tree::types::LayerIterator<K, V>, _>` (opaque type at <../../src/storage/fxfs/src/object_store/tree.rs:27:38>)
              found enum `std::result::Result<impl lsm_tree::types::LayerIterator<K, V>, _>` (trait associated opaque type at <../../src/storage/fxfs/src/object_store/tree.rs:27:38>)
note: required by a bound in `object_store::tree::MajorCompactable::{synthetic#0}`
   |
   |
27 |     ) -> impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send {
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `MajorCompactable::{synthetic#0}`

error: aborting due to 1 previous error



FAILED: gen/src/storage/fxfs/lib.clippy 
../../build/rust/clippy_wrapper.sh --output gen/src/storage/fxfs/lib.clippy --jq ../../prebuilt/third_party/jq/linux-x64/bin/jq --fail -- env  ../../../install/bin/clippy-driver ../../src/storage/fxfs/src/lib.rs --sysroot=../../../install --crate-type=rlib --cfg=__rust_toolchain=\"2024/04/21\ 05:43:23\" -L gen/zircon/public/sysroot/cpp/lib -Clinker=../../prebuilt/third_party/clang/linux-x64/bin/lld -Clink-arg=--sysroot=gen/zircon/public/sysroot/cpp -Clink-arg=-L../../prebuilt/third_party/clang/linux-x64/bin/../lib/x86_64-unknown-fuchsia/noexcept -Clink-arg=-L../../prebuilt/third_party/clang/linux-x64/lib/clang/18/lib/x86_64-unknown-fuchsia -Clink-arg=--pack-dyn-relocs=relr -Clink-arg=-dynamic-linker=ld.so.1 -Clink-arg=--icf=all -Clink-arg=-zrel -Zremap-cwd-prefix=. -Cforce-frame-pointers -Copt-level=0 -Cdebuginfo=0 -Clink-arg=--compress-debug-sections=zstd -Cdebug-assertions=yes -Zallow-features= --target x86_64-fuchsia --cap-lints=warn -Wrust-2018-idioms -Dderef-nullptr -Dinvalid-value -Dunused-must-use -Awhere_clauses_object_safety -Dwarnings -Cprefer-dynamic -Clink-args=-zstack-size=0x200000 -Cpanic=abort -Cforce-unwind-tables=yes -Zpanic_abort_tests -Csymbol-mangling-version=v0 -Wunused_crate_dependencies -Dunused_results -Aunused_results --edition=2021 -Aclippy::all -Dclippy::correctness -Dclippy::missing_safety_doc -Wclippy::expect_fun_call -Wclippy::unnecessary_mut_passed --cfg=fidl_rust_transition_step=\"migrate_all_results\"
error[E0271]: expected `impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send` to be a future that resolves to `Result<impl LayerIterator<K, V>, Error>`, but it resolves to `Result<impl LayerIterator<K, V>, Error>`
   |
   |
27 |     ) -> impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send {
   |
   |
   = note: expected enum `std::result::Result<impl lsm_tree::types::LayerIterator<K, V>, _>` (opaque type at <../../src/storage/fxfs/src/object_store/tree.rs:27:38>)
              found enum `std::result::Result<impl lsm_tree::types::LayerIterator<K, V>, _>` (trait associated opaque type at <../../src/storage/fxfs/src/object_store/tree.rs:27:38>)
note: required by a bound in `object_store::tree::MajorCompactable::{synthetic#0}`
   |
   |
27 |     ) -> impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send {
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `MajorCompactable::{synthetic#0}`

error: aborting due to 1 previous error



FAILED: gen/src/storage/fxfs/_fuzz_journal_lib_rustc_static.clippy 
../../build/rust/clippy_wrapper.sh --output gen/src/storage/fxfs/_fuzz_journal_lib_rustc_static.clippy --jq ../../prebuilt/third_party/jq/linux-x64/bin/jq --fail -- env  ../../../install/bin/clippy-driver ../../src/storage/fxfs/src/lib.rs --sysroot=../../../install --crate-type=staticlib --cfg=fuzz --cfg=fuzz_target=\"fuzz_journal\" -Ccodegen-units=1 --cfg=__rust_toolchain=\"2024/04/21\ 05:43:23\" -L gen/zircon/public/sysroot/cpp/lib -Clinker=../../prebuilt/third_party/clang/linux-x64/bin/lld -Clink-arg=--sysroot=gen/zircon/public/sysroot/cpp -Clink-arg=-L../../prebuilt/third_party/clang/linux-x64/bin/../lib/x86_64-unknown-fuchsia/noexcept -Clink-arg=-L../../prebuilt/third_party/clang/linux-x64/lib/clang/18/lib/x86_64-unknown-fuchsia -Clink-arg=--pack-dyn-relocs=relr -Clink-arg=-dynamic-linker=ld.so.1 -Clink-arg=--icf=all -Clink-arg=-zrel -Zremap-cwd-prefix=. -Cforce-frame-pointers -Copt-level=0 -Cdebuginfo=0 -Clink-arg=--compress-debug-sections=zstd -Cdebug-assertions=yes -Zallow-features= --target x86_64-fuchsia --cap-lints=warn -Wrust-2018-idioms -Dderef-nullptr -Dinvalid-value -Dunused-must-use -Awhere_clauses_object_safety -Dwarnings -Cprefer-dynamic -Clink-args=-zstack-size=0x200000 -Cpanic=abort -Cforce-unwind-tables=yes -Zpanic_abort_tests -Csymbol-mangling-version=v0 -Wunused_crate_dependencies -Dunused_results -Aunused_results --edition=2021 -Aclippy::all -Dclippy::correctness -Dclippy::missing_safety_doc -Wclippy::expect_fun_call -Wclippy::unnecessary_mut_passed --cfg=fidl_rust_transition_step=\"migrate_all_results\"
error[E0271]: expected `impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send` to be a future that resolves to `Result<impl LayerIterator<K, V>, Error>`, but it resolves to `Result<impl LayerIterator<K, V>, Error>`
   |
   |
27 |     ) -> impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send {
   |
   |
   = note: expected enum `std::result::Result<impl lsm_tree::types::LayerIterator<K, V>, _>` (opaque type at <../../src/storage/fxfs/src/object_store/tree.rs:27:38>)
              found enum `std::result::Result<impl lsm_tree::types::LayerIterator<K, V>, _>` (trait associated opaque type at <../../src/storage/fxfs/src/object_store/tree.rs:27:38>)
note: required by a bound in `object_store::tree::MajorCompactable::{synthetic#0}`
   |
   |
27 |     ) -> impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send {
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `MajorCompactable::{synthetic#0}`

error: aborting due to 1 previous error



FAILED: obj/src/storage/fxfs/libfxfs.rlib 
../../prebuilt/third_party/python3/linux-x64/bin/python3 -S ../../build/tracer/restat_cacher.py --outputs obj/src/storage/fxfs/libfxfs.rlib obj/src/storage/fxfs/libfxfs.rlib.d --  RUST_BACKTRACE=1 ../../../install/bin/rustc --color=always --crate-name fxfs ../../src/storage/fxfs/src/lib.rs --crate-type rlib --emit=dep-info=obj/src/storage/fxfs/libfxfs.rlib.d,link -Zdep-info-omit-d-target --cfg=fidl_rust_transition_step=\"migrate_all_results\" --cfg=__rust_toolchain=\"2024/04/21\ 05:43:23\" -L gen/zircon/public/sysroot/cpp/lib -Clinker=../../prebuilt/third_party/clang/linux-x64/bin/lld -Clink-arg=--sysroot=gen/zircon/public/sysroot/cpp -Clink-arg=-L../../prebuilt/third_party/clang/linux-x64/bin/../lib/x86_64-unknown-fuchsia/noexcept -Clink-arg=-L../../prebuilt/third_party/clang/linux-x64/lib/clang/18/lib/x86_64-unknown-fuchsia -Clink-arg=--pack-dyn-relocs=relr -Clink-arg=-dynamic-linker=ld.so.1 -Clink-arg=--icf=all -Clink-arg=-zrel -Zremap-cwd-prefix=. -Cforce-frame-pointers -Copt-level=0 -Cdebuginfo=0 -Clink-arg=--compress-debug-sections=zstd -Cdebug-assertions=yes -Zallow-features= --target x86_64-fuchsia --cap-lints=warn -Wrust-2018-idioms -Dderef-nullptr -Dinvalid-value -Dunused-must-use -Awhere_clauses_object_safety -Dwarnings -Cprefer-dynamic -Clink-args=-zstack-size=0x200000 -Cpanic=abort -Cforce-unwind-tables=yes -Zpanic_abort_tests -Csymbol-mangling-version=v0 -Wunused_crate_dependencies -Dunused_results -Aunused_results --edition=2021 -o obj/src/storage/fxfs/libfxfs.rlib -Ldependency=host_x64/obj/src/storage/fxfs -Ldependency=obj/sdk/fidl/fuchsia.io/fuchsia.io_rust -Ldependency=obj/src/lib/fidl/rust/fidl -Ldependency=obj/src/lib/fuchsia-async -Ldependency=host_x64/obj/src/lib/fuchsia-async-macro -Ldependency=obj/third_party/rust_crates -Ldependency=host_x64/obj/third_party/rust_crates -Ldependency=obj/src/lib/fuchsia-sync -Ldependency=obj/src/lib/zircon/rust -Ldependency=obj/zircon/vdso/zx_zither.rust_syscall -Ldependency=obj/sdk/fidl/fuchsia.unknown/fuchsia.unknown_rust -Ldependency=obj/src/lib/diagnostics/inspect/rust -Ldependency=obj/src/lib/diagnostics/hierarchy/rust -Ldependency=obj/sdk/fidl/fuchsia.diagnostics/fuchsia.diagnostics_rust -Ldependency=obj/sdk/fidl/fuchsia.mem/fuchsia.mem_rust -Ldependency=obj/src/lib/diagnostics/selectors -Ldependency=obj/src/sys/lib/moniker -Ldependency=obj/src/sys/lib/cm_rust -Ldependency=host_x64/obj/src/sys/lib/cm_rust -Ldependency=obj/sdk/fidl/fuchsia.component.decl/fuchsia.component.decl_rust -Ldependency=obj/sdk/fidl/fuchsia.data/fuchsia.data_rust -Ldependency=obj/sdk/fidl/fuchsia.url/fuchsia.url_rust -Ldependency=obj/sdk/fidl/fuchsia.process/fuchsia.process_rust -Ldependency=obj/sdk/fidl/fuchsia.ldsvc/fuchsia.ldsvc_rust -Ldependency=obj/src/lib/flyweights -Ldependency=obj/src/lib/from-enum -Ldependency=host_x64/obj/src/lib/from-enum -Ldependency=obj/src/sys/lib/cm_fidl_validator -Ldependency=obj/src/lib/directed_graph -Ldependency=obj/src/sys/lib/cm_types -Ldependency=obj/src/storage/lib/vfs/rust/name -Ldependency=obj/src/sys/lib/namespace -Ldependency=obj/sdk/fidl/fuchsia.component/fuchsia.component_rust -Ldependency=obj/sdk/fidl/fuchsia.component.sandbox/fuchsia.component.sandbox_rust -Ldependency=obj/sdk/fidl/fuchsia.component.runner/fuchsia.component.runner_rust -Ldependency=obj/sdk/fidl/fuchsia.diagnostics.types/fuchsia.diagnostics.types_rust -Ldependency=obj/src/lib/process_builder -Ldependency=obj/src/lib/fuchsia-runtime -Ldependency=obj/src/storage/lib/vfs/rust -Ldependency=obj/src/storage/lib/trace -Ldependency=host_x64/obj/src/storage/lib/vfs/rust/macros -Ldependency=obj/src/lib/diagnostics/inspect/format/rust -Ldependency=obj/src/lib/injectable-time -Ldependency=obj/sdk/fidl/fuchsia.inspect/fuchsia.inspect_rust -Ldependency=obj/src/lib/mundane -Ldependency=obj/third_party/boringssl/rust/boringssl-sys -Ldependency=obj/src/storage/fxfs/crypto -Ldependency=obj/src/storage/fxfs/trace -Ldependency=host_x64/obj/src/storage/fxfs/trace/macros -Ldependency=obj/src/storage/type-fprint -Ldependency=host_x64/obj/src/storage/type-fprint -Ldependency=obj/src/storage/lib/fsverity-merkle -Ldependency=obj/src/storage/lib/storage_device -Ldependency=obj/src/storage/lib/block_client/rust -Ldependency=obj/sdk/fidl/fuchsia.hardware.block/fuchsia.hardware.block_rust -Ldependency=obj/sdk/fidl/fuchsia.storage.metrics/fuchsia.storage.metrics_rust -Ldependency=obj/sdk/fidl/fuchsia.hardware.block.driver/fuchsia.hardware.block.driver_rust -Ldependency=obj/src/lib/debug/rust -Lnative=obj/zircon/system/ulib/sync -Lnative=x64-shared/link_stub -Lnative=obj/src/lib/debug/rust -Lnative=obj/src/lib/debug -Lnative=obj/zircon/system/ulib/zx -Lnative=obj/sdk/lib/fit -Lnative=obj/sdk/lib/stdcompat -Clink-arg=-Bdynamic -Clink-arg=obj/zircon/system/ulib/sync/libsync.a -Clink-arg=x64-shared/link_stub/libcrypto.so -Clink-arg=obj/src/lib/debug/rust/libdebug_sym_for_rust.a -Clink-arg=obj/src/lib/debug/libdebug.a -Clink-arg=obj/zircon/system/ulib/zx/libzx.a -Clink-arg=obj/sdk/lib/fit/libfit.a -Clink-arg=obj/sdk/lib/stdcompat/libstdcompat.a -Lnative=gen/src/zircon/lib/zircon/zircon.x86_64 -Clink-arg=gen/src/zircon/lib/zircon/zircon.x86_64/libzircon.so --extern fxfs_macros=host_x64/obj/src/storage/fxfs/libfxfs_macros.so --extern fidl_fuchsia_io=obj/sdk/fidl/fuchsia.io/fuchsia.io_rust/libfidl_fuchsia_io.rlib --extern fuchsia_async=obj/src/lib/fuchsia-async/libfuchsia_async.rlib --extern futures=obj/third_party/rust_crates/libfutures-e6516813fcd2ea59.rlib --extern fuchsia_zircon=obj/src/lib/zircon/rust/libfuchsia_zircon.rlib --extern static_assertions=obj/third_party/rust_crates/libstatic_assertions-5fff9c36214c34b7.rlib --extern byteorder=obj/third_party/rust_crates/libbyteorder-eb0270683cf7e0fd.rlib --extern serde=obj/third_party/rust_crates/libserde-703d0860e891586d.rlib --extern scopeguard=obj/third_party/rust_crates/libscopeguard-6ef302a7f864fbb3.rlib --extern rustc_hash=obj/third_party/rust_crates/librustc_hash-d18fd5c05c7b0ca4.rlib --extern thiserror=obj/third_party/rust_crates/libthiserror-dcdc66d3bbb690d3.rlib --extern tracing=obj/third_party/rust_crates/libtracing-979babe725aa0770.rlib --extern once_cell=obj/third_party/rust_crates/libonce_cell-283c53dba5ff9f5a.rlib --extern fuchsia_inspect=obj/src/lib/diagnostics/inspect/rust/libfuchsia_inspect.rlib --extern num_traits=obj/third_party/rust_crates/libnum_traits-eace97eec9f18ef0.rlib --extern anyhow=obj/third_party/rust_crates/libanyhow-99a34437cafed13a.rlib --extern async_trait=host_x64/obj/third_party/rust_crates/libasync_trait-1fd7cf147dfbb38f.so --extern either=obj/third_party/rust_crates/libeither-f4134ec734cf166b.rlib --extern mundane=obj/src/lib/mundane/libmundane.rlib --extern fxfs_crypto=obj/src/storage/fxfs/crypto/libfxfs_crypto.rlib --extern fxfs_trace=obj/src/storage/fxfs/trace/libfxfs_trace.rlib --extern fprint=obj/src/storage/type-fprint/libfprint.rlib --extern fsverity_merkle=obj/src/storage/lib/fsverity-merkle/libfsverity_merkle.rlib --extern storage_device=obj/src/storage/lib/storage_device/libstorage_device.rlib --extern event_listener=obj/third_party/rust_crates/libevent_listener-db0b634d4cbcc40b.rlib --extern rand=obj/third_party/rust_crates/librand-3acdc77c21a3a508.rlib --extern linked_hash_map=obj/third_party/rust_crates/liblinked_hash_map-b9b35caede8d8189.rlib --extern assert_matches=obj/third_party/rust_crates/libassert_matches-568351bf14d9d02.rlib --extern bincode=obj/third_party/rust_crates/libbincode-2294ada8ceec0c61.rlib --extern uuid=obj/third_party/rust_crates/libuuid-10961005f47a1748.rlib --extern debug=obj/src/lib/debug/rust/libdebug.rlib && ../../prebuilt/third_party/python3/linux-x64/bin/python3 -S ../../build/gn/verify_depfile.py -t "//src/storage/fxfs:lib.actual" -d obj/src/storage/fxfs/libfxfs.rlib.d ../../src/storage/fxfs/src/checksum.rs ../../src/storage/fxfs/src/debug_assert_not_too_long.rs ../../src/storage/fxfs/src/drop_event.rs ../../src/storage/fxfs/src/errors.rs ../../src/storage/fxfs/src/filesystem.rs ../../src/storage/fxfs/src/fsck.rs ../../src/storage/fxfs/src/fsck/errors.rs ../../src/storage/fxfs/src/fsck/store_scanner.rs ../../src/storage/fxfs/src/fsck/tests.rs ../../src/storage/fxfs/src/lib.rs ../../src/storage/fxfs/src/log.rs ../../src/storage/fxfs/src/lsm_tree.rs ../../src/storage/fxfs/src/lsm_tree/cache.rs ../../src/storage/fxfs/src/lsm_tree/merge.rs ../../src/storage/fxfs/src/lsm_tree/simple_persistent_layer.rs ../../src/storage/fxfs/src/lsm_tree/skip_list_layer.rs ../../src/storage/fxfs/src/lsm_tree/types.rs ../../src/storage/fxfs/src/metrics.rs ../../src/storage/fxfs/src/object_handle.rs ../../src/storage/fxfs/src/object_store.rs ../../src/storage/fxfs/src/object_store/allocator.rs ../../src/storage/fxfs/src/object_store/allocator/merge.rs ../../src/storage/fxfs/src/object_store/caching_object_handle.rs ../../src/storage/fxfs/src/object_store/data_object_handle.rs ../../src/storage/fxfs/src/object_store/directory.rs ../../src/storage/fxfs/src/object_store/extent_record.rs ../../src/storage/fxfs/src/object_store/flush.rs ../../src/storage/fxfs/src/object_store/graveyard.rs ../../src/storage/fxfs/src/object_store/journal.rs ../../src/storage/fxfs/src/object_store/journal/bootstrap_handle.rs ../../src/storage/fxfs/src/object_store/journal/checksum_list.rs ../../src/storage/fxfs/src/object_store/journal/reader.rs ../../src/storage/fxfs/src/object_store/journal/super_block.rs ../../src/storage/fxfs/src/object_store/journal/writer.rs ../../src/storage/fxfs/src/object_store/key_manager.rs ../../src/storage/fxfs/src/object_store/merge.rs ../../src/storage/fxfs/src/object_store/object_manager.rs ../../src/storage/fxfs/src/object_store/object_record.rs ../../src/storage/fxfs/src/object_store/project_id.rs ../../src/storage/fxfs/src/object_store/store_object_handle.rs ../../src/storage/fxfs/src/object_store/transaction.rs ../../src/storage/fxfs/src/object_store/tree.rs ../../src/storage/fxfs/src/object_store/tree_cache.rs ../../src/storage/fxfs/src/object_store/volume.rs ../../src/storage/fxfs/src/range.rs ../../src/storage/fxfs/src/round.rs ../../src/storage/fxfs/src/serialized_types.rs ../../src/storage/fxfs/src/serialized_types/test_traits.rs ../../src/storage/fxfs/src/serialized_types/tests.rs ../../src/storage/fxfs/src/serialized_types/traits.rs ../../src/storage/fxfs/src/serialized_types/types.rs ../../src/storage/fxfs/src/testing.rs ../../src/storage/fxfs/src/testing/fake_object.rs ../../src/storage/fxfs/src/testing/writer.rs ../../prebuilt/third_party/python3/linux-x64/bin/python3 ../../build/tracer/restat_cacher.py
error[E0271]: expected `impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send` to be a future that resolves to `Result<impl LayerIterator<K, V>, Error>`, but it resolves to `Result<impl LayerIterator<K, V>, Error>`
   |
   |
27 |     ) -> impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send {
   |
   |
   = note: expected enum `Result<impl LayerIterator<K, V>, _>` (opaque type at <../../src/storage/fxfs/src/object_store/tree.rs:27:38>)
              found enum `Result<impl LayerIterator<K, V>, _>` (trait associated opaque type at <../../src/storage/fxfs/src/object_store/tree.rs:27:38>)
note: required by a bound in `MajorCompactable::{synthetic#0}`
   |
   |
27 |     ) -> impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send {
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `MajorCompactable::{synthetic#0}`
error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0271`.
warning: non-local `impl` definition, they should be avoided as they go against expectation
---

warning: 4 warnings emitted


FAILED: gen/src/storage/fxfs/lib_test.clippy 
../../build/rust/clippy_wrapper.sh --output gen/src/storage/fxfs/lib_test.clippy --jq ../../prebuilt/third_party/jq/linux-x64/bin/jq --fail -- env  ../../../install/bin/clippy-driver ../../src/storage/fxfs/src/lib.rs --sysroot=../../../install --crate-type=bin --cfg=__rust_toolchain=\"2024/04/21\ 05:43:23\" -L gen/zircon/public/sysroot/cpp/lib -Clinker=../../prebuilt/third_party/clang/linux-x64/bin/lld -Clink-arg=--sysroot=gen/zircon/public/sysroot/cpp -Clink-arg=-L../../prebuilt/third_party/clang/linux-x64/bin/../lib/x86_64-unknown-fuchsia/noexcept -Clink-arg=-L../../prebuilt/third_party/clang/linux-x64/lib/clang/18/lib/x86_64-unknown-fuchsia -Clink-arg=--pack-dyn-relocs=relr -Clink-arg=-dynamic-linker=ld.so.1 -Clink-arg=--icf=all -Clink-arg=-zrel -Zremap-cwd-prefix=. -Cforce-frame-pointers -Copt-level=0 -Cdebuginfo=0 -Clink-arg=--compress-debug-sections=zstd -Cdebug-assertions=yes -Zallow-features= --target x86_64-fuchsia --cap-lints=warn -Wrust-2018-idioms -Dderef-nullptr -Dinvalid-value -Dunused-must-use -Awhere_clauses_object_safety -Dwarnings -Cprefer-dynamic -Clink-args=-zstack-size=0x200000 -Cpanic=abort -Cforce-unwind-tables=yes -Zpanic_abort_tests -Csymbol-mangling-version=v0 -Wunused_crate_dependencies -Dunused_results -Aunused_results --edition=2021 -Aclippy::all -Dclippy::correctness -Dclippy::missing_safety_doc -Wclippy::expect_fun_call -Wclippy::unnecessary_mut_passed --cfg=fidl_rust_transition_step=\"migrate_all_results\" --test
error[E0271]: expected `impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send` to be a future that resolves to `Result<impl LayerIterator<K, V>, Error>`, but it resolves to `Result<impl LayerIterator<K, V>, Error>`
   |
   |
27 |     ) -> impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send {
   |
   |
   = note: expected enum `std::result::Result<impl lsm_tree::types::LayerIterator<K, V>, _>` (opaque type at <../../src/storage/fxfs/src/object_store/tree.rs:27:38>)
              found enum `std::result::Result<impl lsm_tree::types::LayerIterator<K, V>, _>` (trait associated opaque type at <../../src/storage/fxfs/src/object_store/tree.rs:27:38>)
note: required by a bound in `object_store::tree::MajorCompactable::{synthetic#0}`
   |
   |
27 |     ) -> impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send {
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `MajorCompactable::{synthetic#0}`

error: aborting due to 1 previous error


---

warning: 1 warning emitted


ninja: build stopped: cannot make progress due to previous errors.
error[E0271]: expected `impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send` to be a future that resolves to `Result<impl LayerIterator<K, V>, Error>`, but it resolves to `Result<impl LayerIterator<K, V>, Error>`
   |
   |
27 |     ) -> impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send {
   |
   |
   = note: expected enum `std::result::Result<impl lsm_tree::types::LayerIterator<K, V>, _>` (opaque type at <../../src/storage/fxfs/src/object_store/tree.rs:27:38>)
              found enum `std::result::Result<impl lsm_tree::types::LayerIterator<K, V>, _>` (trait associated opaque type at <../../src/storage/fxfs/src/object_store/tree.rs:27:38>)
note: required by a bound in `object_store::tree::MajorCompactable::{synthetic#0}`
   |
   |
27 |     ) -> impl Future<Output = Result<impl LayerIterator<K, V>, Error>> + Send {
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `MajorCompactable::{synthetic#0}`

warning: method `into_inner` is never used
  --> ../../src/starnix/lib/selinux/policy/src/parser.rs:30:8
   |
```

</details>

--- a/compiler/rustc_hir_analysis/src/collect/item_bounds.rs
+++ b/compiler/rustc_hir_analysis/src/collect/item_bounds.rs
@@ -4,7 +4,7 @@ use rustc_data_structures::fx::FxIndexSet;
 use rustc_hir as hir;
 use rustc_infer::traits::util;
 use rustc_middle::ty::GenericArgs;
-use rustc_middle::ty::{self, Ty, TyCtxt, TypeFoldable, TypeFolder};
+use rustc_middle::ty::{self, Ty, TyCtxt, TypeFoldable, TypeFolder, TypeSuperFoldable};
 use rustc_span::def_id::{DefId, LocalDefId};
 use rustc_span::Span;
 
@@ -214,7 +214,7 @@ impl<'tcx> TypeFolder<TyCtxt<'tcx>> for AssocTyToOpaque<'tcx> {
         {
             self.tcx.type_of(projection_ty.def_id).instantiate(self.tcx, projection_ty.args)
         } else {
-            ty
+            ty.super_fold_with(self)
         }
     }
 }

--- a/tests/ui/impl-trait/in-trait/nested-rpitit-bounds.rs
+++ b/tests/ui/impl-trait/in-trait/nested-rpitit-bounds.rs
@@ -6,6 +6,10 @@ trait Foo {
     fn foo() -> impl Deref<Target = impl Deref<Target = impl Sized>> {
         &&()
     }
+
+    fn bar() -> impl Deref<Target = Option<impl Sized>> {
+        &Some(())
+    }
 }
 
 fn main() {}

